tests/test_minimal/test_tools/test_expand_paths.py

```diff
@@ -380,3 +381,49 @@ def test_expand_paths_ibl(tmpdir):
 
     tc = unittest.TestCase()
     tc.assertCountEqual(path_expansion_results, expected)
+
+
+class TestConstructPathTemplate(unittest.TestCase):
+    def setUp(self):
+        self.path = "/data/subject456/session123/file.txt"
+
+    def test_normal_operation(self):
+        result = construct_path_template(self.path, subject_id="subject456", session_id="session123")
+        self.assertEqual(result, "/data/{subject_id}/{session_id}/file.txt")
+
+    def test_with_additional_metadata(self):
+        result = construct_path_template(
+            "/data/subject789/session456/image.txt", subject_id="subject789", session_id="session456", file_type="image"
```

Comment:
I see what you're checking here, but just my 2 cents: if sticking to `unittest` structure, then would be more consistent to peel this case out into a separate `TestCase`, since the current one mixes application on some on-the-fly cases with the `self.path` example. Just makes it harder to catch if quickly skimming through the testing suite

I'd also say though that `pytest` has been the growing preference preference for new additions to the library, not that that is official policy or anything. Just that it would make this all flatter (all your example paths could be fixtures and each test would be much more obviously unit-like; `pytest.excinfo` is also a bit more convenient to write thorough tests around) and more readable IMO